### PR TITLE
feat(AutoComplete): Expose Input component

### PR DESCRIPTION
**Pull request by zero0Halo**
feat: adds a new prop to AutoComplete, inputComponent. Defaults to Input.

**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [ ] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [ ] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [ ] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [ ] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [ ] Made sure that all accessibility errors are resolved.
- [ ] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [ ] Added name to OWNERS file for all new components
- [ ] If adding a new component, add its export to the rollup config
- [x] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

This adds a new prop, `inputComponent` to `AutoComplete`. It defaults to use the `Input` component, but it allows the user to pass their own `Input` component provided they set up a `forwardRef`. Comments are in the code, documentation updates forthcoming.

This also fixes a rendering issue with Gatsby for `Grid`. Gatsby hates `window` and although I tested for its existence it was a truthiness test and was (apparently) not specific enough.

### CODE REVIEW

File: src/AutoComplete/AutoComplete.component.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ type AutoCompleteDataSource = Array<{
 }>;
 
 interface AutoCompleteProps {
+    inputComponent?: any;
```
Comment (zero0Halo) [inline src/AutoComplete/AutoComplete.component.tsx]: Following the pattern of other Anchor components that accept components as a prop. `ReactElement` throws TS errors, I'm guessing because of the `forwardRef`?

Comment (sharkparty) [reply src/AutoComplete/AutoComplete.component.tsx]: Probably, why not use the `as` prop built into styled-components? I think I'm not up to speed on the ask here, though. Wanna chat about that in person or dump it here?

File: src/AutoComplete/AutoComplete.component.tsx [hunk 2]
```diff
@@ -16,38 +16,39 @@ type AutoCompleteDataSource = Array<{
...
-    border?: boolean;
-    shadow?: boolean;
     background?: string;
+    border?: boolean;
     color?: string;
+    shadow?: boolean;
     // TODO: Allow children
     resultTemplate?: (props: any) => any;
-    suffix?: any;
+    inputComponent?: any;
```
Comment (sharkparty) [inline src/AutoComplete/AutoComplete.component.tsx:36]: Still not clear on this prop. Have you seen [this](https://www.styled-components.com/docs/api#as-polymorphic-prop)?

Comment (zero0Halo) [reply src/AutoComplete/AutoComplete.component.tsx:36]: The `inputComponent` prop allows the user to provide their own JSX to replace the `Input` component being used. Theoretically any JSX could be supplied, but realistically it allows a new `Input` to be supplied with whatever attributes the user wants to tack on. Line `168` has an example of the usage.

I looked into `as` and I'm not exactly sure how that would solve this. I tried `<AutoComplete as="input" autoCapitalize="false />` and get TS errors so I'm guessing that's not what you're suggesting.

Comment (zero0Halo) [reply src/AutoComplete/AutoComplete.component.tsx:36]: I tried using `as` and passing the custom component into it. The problem is that the `ref` gets lost when doing so. I've tried a variety of ways to pass it along including the `forwardedAs` prop from `styled-components` with no luck.
